src/hooks/useFrecency.js

```diff
@@ -0,0 +1,142 @@
+import { useEffect, useState } from 'react';
+import { useAuth } from '../contexts/AuthContextProvider';
+import useLocalStorageState from './useLocalStorageState';
+import { db } from '../utils/FirebaseConfig';
+import { query, collection, onSnapshot, orderBy } from 'firebase/firestore';
+
+const getPartOfDay = (now) => {
+	const hours = now.getHours();
+	let result;
+	switch (hours) {
+		case hours >= 0 && hours <= 5:
+			result = 'night';
+			break;
+		case hours >= 6 && hours <= 11:
+			result = 'morning';
+			break;
+		case hours >= 12 && hours <= 17:
+			result = 'afternoon';
+			break;
+		case hours >= 18 && hours <= 23:
+			result = 'evening';
+			break;
+	}
+
+	return result;
+};
+
+const getPartOfWeek = (now) => {
+	const days = now.getDay();
+	if (days == 0 || days == 6) return 'weekend';
+	return 'weekday';
+};
+
+export default function useFrecency() {
+	const { user } = useAuth();
+	const [history, setHistory] = useLocalStorageState('history', []);
+	const [suggestions, setSuggestions] = useState([]);
+
+	useEffect(() => {
+		if (!user) return;
+
+		const queryRef = query(collection(db, 'users', `${user.uid}/history`), orderBy('visitTime', 'desc'));
+
+		const unsubscribe = onSnapshot(queryRef, (snapshot) => {
+			let historyArray = [];
+			snapshot.forEach((doc) => {
+				const data = doc.data();
+				const historyData = {
+					id: doc.id,
+					visitTime: data.visitTime,
+					title: data.title,
+					url: data.url,
+					favicon: data.favicon,
+				};
+				historyArray.push(historyData);
+			});
+			setHistory(historyArray);
+		});
+
+		return () => unsubscribe();
+	}, [user, setHistory]);
+
+	useEffect(() => {
+		let now = new Date();
+		const partOfDayNow = getPartOfDay(now);
+		const partOfWeekNow = getPartOfWeek(now);
+
+		now = now.getTime();
+		const hour = 1000 * 60 * 60;
+		const day = 24 * hour;
+
+		const filteredHistory = [];
+
+		history.forEach((item) => {
+			if (
+				!item.title?.includes('Google Zoeken') ||
+				(!item.title?.includes('Google Search') && item.visitTime >= now - 30 * day)
+			) {
+				filteredHistory.push(item);
+			}
+		});
+
+		const sortedHistory = filteredHistory.sort(function (a, b) {
+			if (a.title < b.title) {
+				return -1;
+			}
+			if (a.title > b.title) {
+				return 1;
+			}
+			return 0;
+		});
+
+		let splitHistory = sortedHistory.reduce((acc, cur) => {
+			acc[cur.title] = acc[cur.title] || [];
+			acc[cur.title].push(cur);
+			return acc;
+		}, []);
```

Comment:
Wat doet deze reduce precies?

Reply:
Ik kreeg deze comments :	
- Split the history on sorted group
- Creates a new array with objects for every sorted group
- Shape [[Brainhouse: [{...}]],
	      [PXL: [{...},{...}]],...]